xarray/test/test_dataarray.py

```diff
@@ -1407,6 +1407,42 @@ def test_resample_first(self):
                              name='time')
         self.assertDataArrayIdentical(expected, actual)
 
+    def test_resample_first_keep_attrs(self):
+        times = pd.date_range('2000-01-01', freq='6H', periods=10)
+        array = DataArray(np.arange(10), [('time', times)])
+        array.attrs['meta'] = 'data'
+
+        resampled_array = array.resample('1D', dim='time', how='first', keep_attrs=True)
```

Comment:
Wait, I thought the whole point here was that they the different methods (e.g. `mean`, `first`, `max`, ...) are supposed to act in the same way.  I think this is a great place to loop through a series of options:

```Python
for keep_attrs in [True, False]:
    for how in ['mean', 'max', 'min', 'std', 'first', 'last']:
        resampled_array = array.resample('1D', dim='time', how=how, keep_attrs=keep_attrs)

        if keep_attrs:
            # check that attrs are present
        else:
            # check that attrs are not present
```